src/crystal/project/git_dependency.cr

```diff
@@ -0,0 +1,60 @@
+require "crystal/project/dependency"
+require "crystal/project/project_error"
+
+module Crystal
+  class GitDependency < Dependency
+    getter target_dir
+
+    def initialize(@repo, name = nil : String?, branch = nil : String?)
+      unless repo =~ /(.*)(:|\/)(.*)\/(.*)(.git)/
```

Comment:
I'm not exactly sure that this is fool-proof. It works with currently-supported GitHub URL styles, but I'm not sure how much git repository URLs can differ.